### ISSUE TITLE
Obsolete elements are not included in @webref/elements

### ISSUE THREAD

**Issue by foolip** [w3c/webref#445]
Elements listed in https://html.spec.whatwg.org/multipage/obsolete.html are not part of `@webref/elements`, and it would be nice if they were. Maybe with `"obselete": true` in there to allow filtering.

In particular, the fact that `<xmp>` uses `HTMLPreElement` would have been nice to be able to infer from this data.

**Comment by tidoust** [comment]
I don't really see how we could automate the extraction of that information from the HTML spec but both the elements extract and this list are unlikely going to change much, so we could maintain an elements patch to add the info if that seems useful.

I'm not clear how to map obsolete elements to interfaces though. Do the alternatives suggested in the spec necessarily mean that the element uses the alternative interfaces? For instance:
- `applet`: does the element use `HTMLEmbedElement` or `HTMLObjectElement`? Or neither?
- `bgsound`: does the element really use `HTMLAudioElement`?
- `frame`, `frameset`, `noframes`: what interface does each of them use?
- `strike`: `HTMLModElement` (for `<del>`) or `HTMLElement` (for `<s>`)?

**Comment by tidoust** [comment]
> I'm not clear how to map obsolete elements to interfaces though.

I'm still not clear whether the HTML spec mandates an official mapping. I tried to list the right interfaces, defaulting to `HTMLElement`, in the patch proposed in #569, using code such as the following to test whether my assumption was correct:

```js
// The HTML spec says xml must be treated as pre
// This returns true
document.createElement('xmp') instanceof HTMLPreElement

// The HTML spec says plaintext must be treated as pre
// ... but this returns false, so mapping to HTMLElement instead
document.createElement('plaintext') instanceof HTMLPreElement
```

**Comment by foolip** [comment]
Can you try creating the elements by parsing instead of `document.createElement()`? There might be parser aliases confusing matters here.

**Comment by dontcallmedom** [comment]
Here is what I get from parsing HTML obsolete elements:
APPLET HTMLUnknownElement
ACRONYM HTMLElement
BGSOUND HTMLUnknownElement
DIR HTMLDirectoryElement
NOFRAMES HTMLElement
ISINDEX HTMLUnknownElement
KEYGEN HTMLElement
LISTING HTMLPreElement
MENUITEM HTMLUnknownElement
NEXTID HTMLUnknownElement
NOEMBED HTMLElement
PARAM HTMLParamElement
RB HTMLElement
RTC HTMLElement
STRIKE HTMLElement
XMP HTMLPreElement
BASEFONT HTMLElement
BIG HTMLElement
BLINK HTMLUnknownElement
CENTER HTMLElement
FONT HTMLFontElement
MARQUEE HTMLMarqueeElement
MULTICOL HTMLUnknownElement
NOBR HTMLElement
SPACER HTMLUnknownElement
TT HTMLElement
PLAINTEXT HTMLElement

based on
```html
<!doctype html>
<html>
  <meta charset=utf-8>
  <title>Parsing obsolete HTML elements</title>
	  <script>
	    window.addEventListener("DOMContentLoaded", () => 	    document.body.querySelectorAll("*").forEach(n =>

														console.log(n.tagName, n.toString().split(' ')[1].slice(0, -1))));
	  </script>

  <body>
    <applet></applet>
    <acronym></acronym>
<bgsound></bgsound>
  <dir></dir>
  <frame></frame>
  <frameset></frameset>
  <noframes></noframes>
  <isindex></isindex>
  <keygen>
    <listing></listing>
    <menuitem></menuitem>
    <nextid></nextid>
      <noembed></noembed>
      <param>
      
      <rb></rb>
      <rtc></rtc>
      <strike></strike>
      <xmp></xmp>
      <basefont>
      <big></big>
      <blink></blink>
      <center></center>
      <font></font>
      <marquee></marquee>
      <multicol></multicol>
      <nobr></nobr>
	<spacer></spacer>
	  <tt></tt>
<plaintext>

```

**Comment by tidoust** [comment]
Also, the spec actually defines the mapping. It's just hidden in another section, next to the definition of `HTMLUnknownElement`:
https://html.spec.whatwg.org/multipage/dom.html#htmlunknownelement

I'll update the PR to use `HTMLUnknownElement` instead of `HTMLElement` where needed.

**Comment by tidoust** [comment]
Done and released in [`@webref/elements@2.0.0`](https://www.npmjs.com/package/@webref/elements/v/2.0.0)
